File: 56-merge-intervals/merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        sv = sorted(intervals, key= lambda x:x[0])
        prv = sv[0]
        res = []
        for nxt in sv:
            if prv[1] >= nxt[0]:
                prv[1] = max(prv[1], nxt[1])
            else:
                res.append(prv)
                prv = nxt
        res.append(prv)
        return res
        

File: 56-merge-intervals/merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        Arrays.sort(intervals, (a, b) -> a[0] - b[0]);
        List<int[]> lst = new ArrayList<>();
        for(int[] val : intervals) {
            if(lst.isEmpty()) {
                lst.add(val);
            } else {
                int[] lstLast = lst.getLast();
                if (lstLast[1] >= val[0]) {
                    int[] newEnt = new int[] {lstLast[0], Math.max(val[1], lstLast[1])};
                    lst.removeLast();
                    lst.add(newEnt);
                } else {
                    lst.add(val);
                }
            }
        }
        int[][] ans = new int[lst.size()][2];
        int i = 0;
        for(int[] a : lst) {
            ans[i++] = a;
        }
        return ans;
    }
}